ext/symengine/ruby_basic.c

```diff
@@ -32,27 +32,37 @@ VALUE cbasic_binary_op(VALUE self, VALUE operand2,
     sympify(operand2, cbasic_operand2);
 
     cresult = basic_new_heap();
-    cwfunc_ptr(cresult, this, cbasic_operand2);
-    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL, cbasic_free_heap,
-                              cresult);
-    basic_free_stack(cbasic_operand2);
 
+    symengine_exceptions_t error_code
+        = cwfunc_ptr(cresult, this, cbasic_operand2);
+    if (error_code == 0) {
+        result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL,
+                                  cbasic_free_heap, cresult);
+        basic_free_stack(cbasic_operand2);
+    } else {
+        basic_free_stack(cbasic_operand2);
+        raise_exception(error_code);
+    }
     return result;
 }
 
 VALUE cbasic_unary_op(VALUE self,
-                      void (*cwfunc_ptr)(basic_struct *, const basic_struct *))
+                      int (*cwfunc_ptr)(basic_struct *, const basic_struct *))
 {
     basic_struct *this, *cresult;
-    VALUE result;
+    VALUE result = Qnil;
 
     Data_Get_Struct(self, basic_struct, this);
 
     cresult = basic_new_heap();
-    cwfunc_ptr(cresult, this);
-    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL, cbasic_free_heap,
-                              cresult);
 
+    symengine_exceptions_t error_code = cwfunc_ptr(cresult, this);
```

Comment:
You should use one convention. `int` or `symengine_exceptions_t` or `CWRAPPER_OUTPUT_TYPE`. All 3 are used in this PR.